plugins/outputs/application_insights/application_insights.go

```diff
@@ -0,0 +1,410 @@
+	telemetry.Properties = metric.Tags()
+	telemetry.Timestamp = metric.Time()
+
+	var usedFields []string
+	var err error
+
+	// Get the sum of all individual measurements(mandatory property)
+	err = setFloat64TelemetryProperty(&(telemetry.Value), []string{"sum", "value"}, metric, &usedFields)
+	if err != nil {
+		return nil
+	}
+
+	// Get the count of measurements (mandatory property)
+	err = setIntTelemetryProperty(&(telemetry.Count), []string{"count", "samples"}, metric, &usedFields)
+	if err != nil {
+		return nil
+	}
+
+	// We attempt to set Min, Max, StdDev and Variance fields but do not really care if they are not present
+	setFloat64TelemetryProperty(&(telemetry.Min), []string{"min"}, metric, &usedFields)
+	setFloat64TelemetryProperty(&(telemetry.Max), []string{"max"}, metric, &usedFields)
+	setFloat64TelemetryProperty(&(telemetry.StdDev), []string{"stddev"}, metric, &usedFields)
+	setFloat64TelemetryProperty(&(telemetry.Variance), []string{"variance"}, metric, &usedFields)
+
+	addRemainingMeasurements(telemetry, metric, usedFields)
+
+	return telemetry
+}
+
+func setFloat64TelemetryProperty(
+	telemetryProperty *float64,
+	candidateFields []string,
+	metric telegraf.Metric,
+	usedFields *[]string) error {
+
+	for _, fieldName := range candidateFields {
+		if metric.HasField(fieldName) {
+			metricValue, err := toFloat64(metric.Fields()[fieldName])
+			if err != nil {
+				return err
+			}
+
+			*telemetryProperty = metricValue
+
+			if usedFields != nil {
+				*usedFields = append(*usedFields, fieldName)
+			}
+
+			return nil
+		}
+	}
+
+	return fmt.Errorf("No field from the candidate list was found in the metric")
+}
+
+func setIntTelemetryProperty(
+	telemetryProperty *int,
+	candidateFields []string,
+	metric telegraf.Metric,
+	usedFields *[]string) error {
+
+	for _, fieldName := range candidateFields {
+		if metric.HasField(fieldName) {
+			metricValue, err := toInt(metric.Fields()[fieldName])
+			if err != nil {
+				return err
+			}
+
+			*telemetryProperty = metricValue
+
+			if usedFields != nil {
+				*usedFields = append(*usedFields, fieldName)
+			}
+
+			return nil
+		}
+	}
+
+	return fmt.Errorf("No field from the candidate list was found in the metric")
+}
+
+func addRemainingMeasurements(telemetry appinsights.Telemetry, metric telegraf.Metric, usedFields []string) {
+	properties := telemetry.GetProperties()
+
+	for fieldName, fieldValue := range metric.Fields() {
+		if contains(usedFields, fieldName) {
+			continue
+		}
+
+		float64Value, err := toFloat64(fieldValue)
+		if err != nil {
+			continue
+		}
+
+		// We would like to add the field to telemetry.Measurements here
+		// But because of issue https://github.com/Microsoft/ApplicationInsights-Go/issues/12
+		// this is currently impossible. So we are just going to add the measurement as a property instead.
+		// This is what we want to be able to do, eventually:
+		//     telemetry.GetMeasurements()[fieldName] = float64Value
```

Comment:
I assume that `telemetry` is always a metric (whether single or multiple value)?

This is unfortunate.  We lose many possible ways to analyze these measurements by stringifying them into properties (which, btw, I'm not sure you can even see except in the AA portal)

Is it really not feasible to log these as separate metrics?

Reply:
It is, will change that